File: Rubyとか/12_EuclideanAlgorithm_ChocolatesByNumbers.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"


def GCD(m, n)
	return 0 if m == 0 || n == 0
	while m != 0 && n != 0
		if m > n then
			m %= n
		else
			n %= m
		end
	end
	return [m, n].max
end

def LCM(m, n)
	return 0 if m == 0 || n == 0
	return m * n / GCD(m, n)
end


def solution(n, m)
  # write your code in Ruby 2.2
  return n / GCD(n, m)
end

File: Python/12_EuclideanAlgorithm_ChocolatesByNumbers.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

# ユークリッド互除法で最大公約数
# pythonだとmath.gcd()とかあるけどね
def gcd(a, b):
    if a * b == 0: return 0
    while a * b != 0:
        a, b = max(a, b), min(a, b)
        a = a % b 
    return int(max(a, b))
# end of func 

def lcm(a, b):
    if a * b == 0: return 0 
    return int(a * b / gcd(a, b))
# end of lcm

def solution(N, M):
    # write your code in Python 3.6
    
    # (M * a) mod N はすべてGCD(M, N)の倍数である
    # GCD(M, N) * a <= N となるaをすべて列挙すれば良いことになる
    return N // gcd(N, M)
# end of func